Sources/TuistCore/ValueGraph/ValueGraphTraverser.swift

```diff
@@ -8,13 +8,50 @@ public class ValueGraphTraverser: GraphTraversing {
     public var path: AbsolutePath { graph.path }
     public var workspace: Workspace { graph.workspace }
     public var projects: [AbsolutePath: Project] { graph.projects }
+    public var targets: [AbsolutePath: [String: Target]] { graph.targets }
+    public var dependencies: [ValueGraphDependency: Set<ValueGraphDependency>] { graph.dependencies }
 
     private let graph: ValueGraph
 
     public required init(graph: ValueGraph) {
         self.graph = graph
     }
 
+    public func rootTargets() -> Set<ValueGraphTarget> {
+        Set(targets.flatMap { (path: AbsolutePath, targets: [String: Target]) -> [ValueGraphTarget] in
+            guard path == graph.path else { return [] }
+            guard let project = projects[path] else { return [] }
+            return targets.values.map { ValueGraphTarget(path: path, target: $0, project: project) }
+        })
+    }
+
+    public func rootProject() -> Project? {
+        projects[graph.path]
+    }
```

Comment:
The semantics here are slightly different, previously a graph had an array of `entryNodes` which could point to multiple projects - e.g. in a workspace the root path may not point to an individual project.

```
Root              <--- `graph.path`
- Workspace.swift
  - App
     - Project.swift
  - Modules
     - MyFramework
        - Project.swift
```

```swift
Workspace(
   // ...
   projects: ["App", "Modules/MyFramework"]
)
```

In such case:
- `graph.path` == `"/Root"`
- `rootProject()` == `nil`

To preserve the previous semantics, we can leverage `workspace.projects`

e.g.

```swift
   func rootTargets() -> Set<ValueGraphTarget> {
        graph.workspace.projects.reduce(into: Set()) { (result, path) in
            result.formUnion(targets(at: path))
        }
    }
    
    func rootProjects() -> [Project] {
        graph.workspace.projects.compactMap {
            projects[$0]
        }
    }
```